File: app/src/main/java/com/hkb48/keepdo/com/hkb48/keepdo/util/DateComparator.java
package com.hkb48.keepdo.com.hkb48.keepdo.util;

import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;

public class DateComparator {
    public static boolean equals(Date a, Date b) {
        Date compA = truncate(a);
        Date compB = truncate(b);

        return compA.equals(compB);
    }

    private static Date truncate(Date datetime) {
        Calendar cal = Calendar.getInstance();
        cal.setTime(datetime);

        return new GregorianCalendar(cal.get(Calendar.YEAR), cal.get(Calendar.MONTH), cal.get(Calendar.DATE))
                .getTime();
    }
}

File: app/src/main/java/com/hkb48/keepdo/util/DateComparator.kt
package com.hkb48.keepdo.util

import java.util.*

object DateComparator {
    fun equals(a: Date?, b: Date?): Boolean {
        if (a == null || b == null) {
            return false
        }
        return truncate(a) == truncate(b)
    }

    private fun truncate(datetime: Date): Date {
        val cal = Calendar.getInstance()
        cal.time = datetime
        return GregorianCalendar(cal[Calendar.YEAR], cal[Calendar.MONTH], cal[Calendar.DATE])
            .time
    }
}